experimental/include/experimental/Transforms/Speculation/SpeculationPlacement.h

```diff
@@ -0,0 +1,109 @@
+//===- SpeculationPlacement.h - Speculation units placement -----*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the data structures needed for speculation algorithms
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef DYNAMATIC_TRANSFORMS_SPECULATION_PLACEMENT_H
+#define DYNAMATIC_TRANSFORMS_SPECULATION_PLACEMENT_H
+
+#include "circt/Dialect/Handshake/HandshakeOps.h"
+#include "dynamatic/Support/DynamaticPass.h"
+#include "dynamatic/Support/LLVM.h"
+#include "dynamatic/Support/Logging.h"
+#include "dynamatic/Support/TimingModels.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Pass/Pass.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/Hashing.h"
+#include <map>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace dynamatic {
+namespace experimental {
+namespace speculation {
+
+struct PlacementOperand {
+  StringRef opName;
```

Comment:
Wondering whether it is safe to store only a reference to the string here memory-wise. Do you have the guarantee that the underlying memory region holding the string is still there when you end up using the struct? If the string was parsed from the JSON file, I don't think it would be. If that's true, replace this with a `std::string`.

Reply:
Replaced with `std::string` because I agree that the problem that you are mentioning could happen.